File: Pre_requisites/Patterns/cpp/pattern_7.cpp
/*
Problem statement:
Ninja was very fond of patterns. For a given integer ‘N’, he wants to make the N-Star Triangle.

Example:
Input: ‘N’ = 3

Output: 

  *
 ***
*****

*/

void nStarTriangle(int n) {
    
    int i,j,k,n_spaces,n_stars;

    for(i=1;i<n+1;i++){

        n_stars = 2*i-1;
        n_spaces = n-i;

        for(j=1;j<=n_spaces;j++){
            cout<<" ";
        }

        for(k=1;k<=n_stars;k++){
            cout<<"*";
        }
        cout<<endl;
    
    }
}

File: Pre_requisites/Patterns/python/pattern_7.py
'''
Problem statement:
Ninja was very fond of patterns. For a given integer ‘N’, he wants to make the N-Star Triangle.

Example:
Input: ‘N’ = 3

Output: 

  *
 ***
*****

'''

def nStarTriangle(n: int) -> None:
    
    for i in range(1,n+1):

        n_stars=2*i-1
        n_spaces=n-i

        print(" "*n_spaces+"*"*n_stars)